File: C++/programmers_12909.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    stack<char> st;
    for (auto it : s) {
        if (it == '(') {
            st.push(it);
        }
        else {
            if (st.size() == 0 || st.top() != '(') {
                return false;
            }
            st.pop();
        }
    }

    // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    cout << "Hello Cpp" << endl;

    if (st.size() != 0)
        return false;

    return answer;
}

File: java/programmers_12909.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        
        for(int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            
            if(c == '(') {
                stack.push(c);
            }else {
                if(stack.empty() || stack.peek() != '(') {
                    answer = false;
                    break;
                }else {
                    stack.pop();
                }
            }
        }
        
        if(!stack.empty()) {
            answer = false;
        }

        return answer;
    }
}